### PR TITLE
Fix ocm end-point discovery discovery

### LINKED ISSUE #14247: ocm-provider end-point doesn't work
Tested with Nextcloud 15.0.4

````
curl -X GET http://localhost:70/server/ocm-provider/
````

should return a JSON object which describes the OCM capabilities. This works on my dev environment but not on productive instances like c.nc.com. I assume that it doesn't work because of the rewrite rules in the .htaccess file. But `/ocs-provider/` works and there is no special rule for this end-point in the .htaccess and I can't spot a obvious reason why it doesn't work for `/ocm-provider/`

@rullzer @MorrisJobke any idea?

### PR DESCRIPTION
**Pull request by schiessle**
fix https://github.com/nextcloud/server/issues/14247

### CODE REVIEW

**Comment by rullzer** [comment]
TODO:
- [ ] Update nginx docs

**Comment by schiessle** [comment]
@rullzer updated the documentation as well: https://github.com/nextcloud/documentation/pull/1260

**Comment by rullzer** [comment]
@MorrisJobke just editing it there is enough to also trigger the recreation of htacceess during update right?

**Comment by MorrisJobke** [comment]
Please add a setup check to the admin settings to notify users about this.

See for example the well-known check in the browser that does something like this:

https://github.com/nextcloud/server/blob/813ff430f159fdc8aade236048f4a8ff18ac3b2e/settings/js/admin.js#L251

**Comment by MorrisJobke** [comment]
> @MorrisJobke just editing it there is enough to also trigger the recreation of htacceess during update right?

It is written on every update - so yes.

**Comment by schiessle** [comment]
I added a check for both (while I'm on it), the ocs-provider and the ocm-provider route, see the last two commits

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -91,6 +91,46 @@
...
+			var deferred = $.Deferred();
+
+			if(runCheck === false) {
+				deferred.resolve([]);
+				return deferred.promise();
+			}
+			var afterCall = function(xhr) {
+				var messages = [];
+				if (expectedStatus.indexOf(xhr.status) === -1) {
+					var docUrl = placeholderUrl.replace('PLACEHOLDER', 'admin-setup-well-known-URL');
```
Comment (MorrisJobke) [inline core/js/setupchecks.js]: 🙈 This is not the correct doc. Let me fix that.

File: settings/js/admin.js [hunk 2]
```diff
@@ -250,6 +250,8 @@ $(document).ready(function(){
 		OC.SetupChecks.checkWellKnownUrl('/.well-known/webfinger', oc_defaults.docPlaceholderUrl, $('#postsetupchecks').data('check-wellknown') === true && !!oc_appconfig.core.public_webfinger, [200, 501]),
 		OC.SetupChecks.checkWellKnownUrl('/.well-known/caldav', oc_defaults.docPlaceholderUrl, $('#postsetupchecks').data('check-wellknown') === true),
 		OC.SetupChecks.checkWellKnownUrl('/.well-known/carddav', oc_defaults.docPlaceholderUrl, $('#postsetupchecks').data('check-wellknown') === true),
+		OC.SetupChecks.checkProviderUrl('/ocm-provider/', oc_defaults.docPlaceholderUrl, $('#postsetupchecks').data('check-wellknown') === true),
+		OC.SetupChecks.checkProviderUrl('/ocs-provider/', oc_defaults.docPlaceholderUrl, $('#postsetupchecks').data('check-wellknown') === true),
 		OC.SetupChecks.checkSetup(),
 		OC.SetupChecks.checkGeneric(),
 		OC.SetupChecks.checkWOFF2Loading(OC.filePath('core', '', 'fonts/Nunito-Regular.woff2'), oc_defaults.docPlaceholderUrl),
```
Comment (MorrisJobke) [inline settings/js/admin.js]: Also then the list below with `check1, check2,...` should be extended

Comment (MorrisJobke) [reply settings/js/admin.js]: I will also fix that and fix the JSUnit tests.

**Comment by MorrisJobke** [comment]
And let me also rebase to resolve conflicts ...

**Comment by schiessle** [comment]
@MorrisJobke thanks for taking care of it! :+1:

**Comment by MorrisJobke** [comment]
And let me add some unit tests for it.

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Backport in #14312

**Comment by MorrisJobke** [comment]
@schiessle Is there any reason why this is not put under the `/.well-known/` path, because this is the exact purpose of this?

**Pull request merged**